api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -157,13 +130,8 @@ private String convertClusterName(String workspaceId) {
       String clusterName = convertClusterName(workspaceId);
       this.notebooksService.localize(workspaceNamespace, clusterName,
           convertfileDetailsToMap(fileList));
-    } catch (ApiException e) {
-      if (e.getCode() == 400) {
-        throw new BadRequestException(e.getResponseBody());
-      } else if (e.getCode() == 404) {
-        throw new NotFoundException("Cluster not found.");
-      }
-      throw new ServerErrorException(e);
+    } catch (NotFoundException e) {
+      throw new NotFoundException("Cluster not found.");
```

Comment:
Just for my own information, what is the purpose of catching and rethrowing here?

Reply:
So the generic notebook `ApiException` converter will just yield an empty `NotFoundExeption()`; the idea is that in some cases, depending on what we are trying to do, we might want to catch and rethrow with a more specific message to the user.

Reexamining this case though, I don't think we have any more information at this level than we do in `NotebookServiceImpl`; and I'm also not entirely convinced that this is the only scenario in which Leo would return a 404, so I've just dropped this handling for now. I think the other usage in this PR in `WorkspaceController` is probably OK, as it is otherwise ambiguous as to what is missing (the workspace? the files? the cluster?).